api/proxy/store/generated_key/memstore/v2/memstore.go

```diff
@@ -180,39 +181,63 @@ func (e *MemStore) generateRandomCert(blobContents []byte) (coretypes.EigenDACer
 func (e *MemStore) Get(
 	_ context.Context, versionedCert certs.VersionedCert, returnEncodedPayload bool,
 ) ([]byte, error) {
-	encodedBlob, err := e.FetchEntry(crypto.Keccak256Hash(versionedCert.SerializedCert).Bytes())
+	blobSerialized, err := e.FetchEntry(crypto.Keccak256Hash(versionedCert.SerializedCert).Bytes())
+	if err != nil {
+		return nil, fmt.Errorf("fetching entry via memstore: %w", err)
+	}
+
+	var v3cert coretypes.EigenDACertV3
+	err = rlp.DecodeBytes(versionedCert.SerializedCert, &v3cert)
+	if err != nil {
+		return nil, coretypes.ErrCertParsingFailedDerivationError
+	}
+
+	blob, err := coretypes.DeserializeBlob(
+		blobSerialized,
+		v3cert.BlobInclusionInfo.BlobCertificate.BlobHeader.Commitment.Length,
+	)
 	if err != nil {
-		return nil, fmt.Errorf("fetching entry via v2 memstore: %w", err)
+		return nil, fmt.Errorf("deserialize blob: %w", err)
 	}
 
 	if returnEncodedPayload {
-		return encodedBlob, nil
+		encodedPayload := blob.ToEncodedPayloadUnchecked(e.polyForm)
+		return encodedPayload.Serialize(), nil
 	}
 
-	return e.codec.DecodeBlob(encodedBlob)
+	payload, err := blob.ToPayload(e.polyForm)
+	if err != nil {
+		return nil, fmt.Errorf("convert blob to payload: %w", err)
+	}
+	return payload, nil
 }
 
 // Put inserts a value into the store.
 // ephemeral db key = keccak256(pseudo_random_cert)
 // this is done to verify that a rollup must be able to provide
 // the same certificate used in dispersal for retrieval
 func (e *MemStore) Put(_ context.Context, value []byte) ([]byte, error) {
-	encodedVal, err := e.codec.EncodeBlob(value)
+	payload := coretypes.Payload(value)
+
+	blob, err := payload.ToBlob(e.polyForm)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("generating blob: %w", err)
 	}
 
-	artificialV2Cert, err := e.generateRandomCert(encodedVal)
+	blobSerialized := blob.Serialize()
+
+	// generateRandomV3Cert produces valid kzg commitment
+	artificialV3Cert, err := e.generateRandomV3Cert(blobSerialized)
 	if err != nil {
 		return nil, fmt.Errorf("generating random cert: %w", err)
 	}
 
-	certBytes, err := rlp.EncodeToBytes(artificialV2Cert)
+	certBytes, err := rlp.EncodeToBytes(artificialV3Cert)
 	if err != nil {
 		return nil, fmt.Errorf("rlp decode v2 cert: %w", err)
 	}
 
-	err = e.InsertEntry(crypto.Keccak256Hash(certBytes).Bytes(), encodedVal)
+	err = e.InsertEntry(crypto.Keccak256Hash(certBytes).Bytes(), blobSerialized)
```

Comment:
just realized that we effectively hard-fork memstore every time we introduce a new cert version. for testability it might be beneficial to support this within memconfig going forward for when we need to test cross-compatibility between multiple cert versions in a rollup derivation pipeline. def out-of-scope of this PR though